parla/task_runtime.py

```diff
@@ -1804,6 +1797,19 @@ def _map_tasks(self):
             task: Optional[Task] = self._dequeue_spawned_task()
             if task:
                 if not task.assigned:
+                    _new_launched_task = []
+                    for n_task in self._launched_tasks:
+                        task_state = n_task.state
+                        if isinstance(task_state, TaskCompleted):
+                            for dev in n_task.req.devices:
+                                self._available_resources.deallocate_resources(dev, n_task.req.resources)
```

Comment:
I think that this doesn't handle Data Movement Tasks? Their resources are still bundled with their "parents", but their resources are no longer allocated 'ahead' of time. Only compute tasks are being tracked and deallocated. This means that they currently are considered to take 'no resources' and can be over-scheduled.

Reply:
Although (sorry for the stream of consciousness comments on this), doesn't this mean it can still deadlock in the same way as the current mapper? Since it can allocate a bunch of data movement tasks out of order and prevent the next dependency from being mapped due to being out of resources?

Reply:
Yes. still deadlock happens. Ok, let's consider this PR as the immediate solution for the performance. I think your pointing out could not be resolved by this PR. Let's consider that issue on the separate PR with more discussion.